contacts/tests.py

```diff
@@ -1,3 +1,123 @@
-# from django.test import TestCase
+import pytest
+from users.models import User, Seeker
+from apartments.models import Apartment, City
+from .models import Connection
+from django.db import transaction, IntegrityError
 
-# Create your tests here.
+
+@pytest.fixture
+def new_city():
+    city = City(cityName='Tel Aviv')
+    city.save()
+    return city
+
+
+@pytest.fixture
+def seeker_1(new_city):
+    user1 = create_new_user("t1@m.com", "test1", "test", "1995-05-05", "testing")
+    seeker = create_seeker(user1, new_city, "2020-05-05", 100, 3000, 2, 2, "Hi")
+    seeker.save()
+    return seeker
+
+
+@pytest.fixture
+def seeker_2(new_city):
+    user2 = create_new_user("t2@m.com", "test2", "test", "1995-05-05", "testing")
+    seeker = create_seeker(user2, new_city, "2020-05-05", 300, 3000, 3, 3, "Hi")
+    seeker.save()
+    return seeker
+
+
+@pytest.fixture
+def apartment_1(new_city):
+    user3 = create_new_user("t3@m.com", "test3", "test", "1995-05-05", "testing")
+    apartment = create_apartment(user3, new_city, "Hatotahim 5", 2500, 2, 3, "2020-05-05")
+    apartment.save()
+    return apartment
+
+
+@pytest.fixture
+def connection_1(seeker_1, apartment_1):
+    con = Connection(seeker=seeker_1, apartment=apartment_1)
+    con.save()
+    return con
+
+
+@pytest.mark.django_db
+def test_add_connection(connection_1):
+    assert connection_1 is not None
+    assert connection_1.seeker.base_user.email == "t1@m.com"
+    assert connection_1.apartment.owner.email == "t3@m.com"
+
+
+@pytest.mark.django_db
+def test_find_connection(connection_1):
+    temp_connection = Connection.objects.get(seeker=connection_1.seeker, apartment=connection_1.apartment)
+    assert temp_connection == connection_1
+
+
+@pytest.mark.django_db
+def test_invalid_users_connection(seeker_1, seeker_2):
+    with pytest.raises(ValueError):
+        with transaction.atomic():
+            Connection(seeker=seeker_1, apartment=seeker_2).save()
+
+
+@pytest.mark.django_db
+def test_invalid_duplicate_connection(connection_1, seeker_1, apartment_1):
+    with pytest.raises(IntegrityError):
+        with transaction.atomic():
+            Connection(seeker=seeker_1, apartment=apartment_1).save()
```

Comment:
This is a bit ambiguous. You send connection_1 to the function but don't use it, I think I know why you are doing just that but I would like to hear the reason please.

Reply:
@1Oreo Connection_1 is created as a pytest fixture, it saves a row in the db with a connection between seeker_1 and apartment_1.
In the test, I want to save *another* connection row between seeker_1 and apartment_1, which should cause an IntegrityError. To make that first original connection be inserted to the db I had to send connection_1 to the function. It's a fixture so as long as I don't add it as an argument, it's like that row was never added in the first place (will not activate the fixture and therefore will not save connection_1). Causing the "second" row that was added in the test function to be the first and only one, which will not cause a duplication error.

Reply:
So I figured this is what you meant to create.
But the way you implemented it might cause someone to misunderstand the intention. What I would do to sort it out is use `connection_1.seeker` and `connection_1.appartment` so that when someone reads the code he will understand that you are re-inserting the same connection to cause an error.

Reply:
@1Oreo You got me 🎯 
I took your tip and changed it, let me know what you think :)